olive/systems/azureml/aml_system.py

```diff
@@ -115,14 +117,78 @@ def __init__(
         self.env_vars = self._get_hf_token_env(self.azureml_client_config.keyvault_name) if self.hf_token else None
         self.temp_dirs = []
 
+    def submit_workflow(self, run_config: RunConfig):
+        workflow_id = run_config.workflow_id
+        logger.info("Submitting workflow %s to the AzureML system.", workflow_id)
+        ml_client = self.azureml_client_config.create_client()
+
+        with tempfile.TemporaryDirectory() as tempdir:
+            workflow_pipeline = self._create_pipeline_for_workflow(run_config, tempdir)
+            self._run_workflow_job(
+                ml_client,
+                workflow_pipeline,
+                "olive-workflow",
+                tags={"Workflow": workflow_id},
+            )
+
+    def _create_pipeline_for_workflow(self, run_config: RunConfig, tmp_dir):
+        workflow_id = run_config.workflow_id
+        logger.info("Creating pipeline for workflow %s", workflow_id)
+        script_name = "aml_workflow_runner.py"
+
+        tmp_dir = Path(tmp_dir)
+
+        run_config.workflow_host = None
+
+        config_root = tmp_dir / "config"
+        cur_dir = Path(__file__).resolve().parent
+        code_files = [cur_dir / script_name]
+        self.copy_files(code_files, config_root)
+
+        inputs, args = self.create_inputs_and_args(
+            {WORKFLOW_CONFIG: run_config.to_json()},
```

Comment:
nit: since the run_config is already is validated, should we pop `systems`, `evaluators`, `data_configs`,  from the run_config json first? Those are not used at all and complicate the config as well as resources. For example, I see that the aml system related resources are uploaded in your test job.

Reply:
What do you mean by they are not used? The remote workflow can also run evaluation and submit a job to aml system (though not verified yet, will verify this later).

Reply:
They are not used since their values are already inserted directly into engine and passes during the runconfig validation. So, there is nothing in the workflow referring to these anymore.

Reply:
Updated. I will merge after your AML client insertion PR.